packages/core/src/lib/StreamName.ts

```diff
@@ -12,6 +12,11 @@ export function parseCategory(streamName: string) {
   return streamName.slice(0, idx)
 }
 
+export function parseCategoryAndIds(streamName: string) {
```

Comment:
parseCategoryAndId at L5 should have a diff name to signify it does not parse and only splits
Also if there is no cat separator, does/should it throw or return undefined?

L1 compose is also problematic
- ideally category should be checked for embedded `-`
- and create is a better name as it just really glues them together

compose should string N elements together and add `_` seps (checking for embedded `_` in values)
- maybe it can be called join and parseIds can become split
- or is this stuff all called gen and the opposite is called parse?

Having a different name sets different expectations

Maybe doing a 3 liner intro at top of file about gen vs parse and create vs split or something might guide the naming?

Then, when parsing, there should be:
- Inverse of create (something like split?) to yield a pair
- the thing that gets the category (parseCategory) could just be called category as it does not parse
- the thing that gets the streamId (parseId) could be called streamId as it does not parse
- inverse of compose is called parseIds, maybe it should be called split (esp if you add a join)